Source/FakeItEasy/Core/FakeScope.cs

```diff
@@ -13,19 +16,48 @@ namespace FakeItEasy.Core
     internal abstract class FakeScope
         : IFakeScope
     {
+        private const string CurrentScopeLogicalCallContextKey = "FakeItEasy.FakeScope.Current";
+
+        private static readonly RootScope root;
+        private static readonly ConcurrentDictionary<Guid, FakeScope> scopes;
+
+        [SuppressMessage(
+            "Microsoft.Performance", 
+            "CA1810:InitializeReferenceTypeStaticFieldsInline", 
+            Justification = "RootScope constructor requires scopes to be initialized first.")]
         static FakeScope()
         {
-            Current = new RootScope();
+            scopes = new ConcurrentDictionary<Guid, FakeScope>();
+            root = new RootScope();
         }
 
         private FakeScope()
         {
+            this.Id = Guid.NewGuid();
+            FakeScope.scopes.TryAdd(this.Id, this);
         }
 
-        internal static FakeScope Current { get; set; }
+        internal static FakeScope Current
+        {
+            get
+            {
+                var id = (Guid?)CallContext.LogicalGetData(CurrentScopeLogicalCallContextKey);
+                FakeScope scope;
+                return id.HasValue && FakeScope.scopes.TryGetValue(id.Value, out scope) ? scope : FakeScope.root;
+            }
+            set
+            {
+                if (value == null)
+                    throw new ArgumentNullException("value");
+
+                CallContext.LogicalSetData(CurrentScopeLogicalCallContextKey, value.Id);

```

Comment:
Why use LogicalSetData and not SetData? I would not think you would want the current fake scope to propagate with remoting calls, since the fake scope cannot be shared across app domains.